site/src/pages/docs/server-timeouts.mdx

```diff
@@ -0,0 +1,210 @@
+```
+
+## Using <type://VirtualHostBuilder>
+
+You can set the timeouts for each virtual host individually
+
+```java
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+
+import java.time.Duration;
+
+import static com.linecorp.armeria.common.HttpStatus.OK;
+
+int requestTimeout1 = 5;
+int requestTimeout2 = 10;
+int requestTimeout3 = 20;
+
+ServerBuilder sb = Server.builder();
+Server server = sb.http(port)
+                  .defaultVirtualHost()
+                  .service("/hello1", (ctx, req) -> HttpResponse.of(OK))
+                  .requestTimeout(Duration.ofSeconds(requestTimeout1))
+                  .and()
+                  .virtualHost("*.foo.com")
+                  .service("/hello2", (ctx, req) -> HttpResponse.of(OK))
+                  .requestTimeout(Duration.ofSeconds(requestTimeout2))
+                  .and()
+                  .withVirtualHost(builder -> {
+                      builder.hostnamePattern("*.bar.com")
+                              .service("/hello3", (ctx, req) -> HttpResponse.of(OK))
+                              .requestTimeout(Duration.ofSeconds(requestTimeout3));
+                  })
+                  .build();
+```
+
+## Using <type://ServiceBindingBuilder>
+
+You can set the timeouts for each route individually
+
+```java
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+
+import java.time.Duration;
+
+import static com.linecorp.armeria.common.HttpStatus.OK;
+
+int requestTimeout1 = 5;
+int requestTimeout2 = 10;
+
+ServerBuilder sb = Server.builder();
+Server server = sb.http(port)
+                  .route()
+                  .get("/hello1")
+                  .requestTimeout(Duration.ofSeconds(requestTimeout1))
+                  .build((ctx, req) -> HttpResponse.of(OK))
+                  .withRoute(builder -> {
+                      builder.get("/hello2")
+                             .requestTimeout(Duration.ofSeconds(requestTimeout2))
+                             .build((ctx, req) -> HttpResponse.of(OK));
+                  })
+                  .build();
+```
+
+## Using <type://ServiceRequestContext>
+
+You can use the context to set the timeouts for a request.
+
+```java
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+
+import java.time.Duration;
+
+import static com.linecorp.armeria.common.HttpStatus.OK;
+
+int requestTimeout = 5;
+
+ServerBuilder sb = Server.builder();
+Server server = sb.http(port)
+                  .service("/hello", (ctx, req) -> {
+                      ctx.setRequestTimeout(Duration.ofSeconds(requestTimeout));
+                      return HttpResponse.of(OK);
+                  })
+                  .build();
+```
+
+Also, It is possible to use automatically injected context.
```

Comment:
Optional) Personally, I find this example superfluous since it describes that `ServiceRequestContext` can be injected rather than focusing on setting the timeout. What do you think of removing L155-L184? 
I'm fine either way though

Reply:
I agree with your opinion 👍